src/main/java/seedu/mtracker/commands/CheckOffInstrumentCommand.java

```diff
@@ -0,0 +1,31 @@
+package seedu.mtracker.commands;
+
+import seedu.mtracker.error.ErrorMessage;
+import seedu.mtracker.ui.TextUi;
+
+public class CheckOffInstrumentCommand extends Command {
+
+    public int completedInstrumentIndex;
+    public static final String COMMAND_WORD = "done";
+
+    public CheckOffInstrumentCommand(int completedInstrumentIndex) {
+        this.completedInstrumentIndex = completedInstrumentIndex;
+    }
+
+    public boolean isValidIndex(int completedInstrumentIndex) {
+        boolean isPositive = completedInstrumentIndex >= 0;
+        boolean isLessThanListSize = completedInstrumentIndex <= instrumentManager.getSize() - 1;
+        return isPositive && isLessThanListSize;
+    }
```

Comment:
I think theres no need to create this function here. Try to leave all validation checking related stuff to the parser. Can look at my delete PR

Reply:
I have updated my delete PR to reflect more of how the error handling shld be done